tests/flavor_test.go

```diff
@@ -353,6 +357,147 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 			for _, disk := range vmi.Spec.Domain.Devices.Disks {
 				Expect(disk.DiskDevice.Disk.Bus).To(Equal(preference.Spec.Devices.PreferredDiskBus))
 			}
+		})
+
+		It("[test_id:TODO] should store and use ControllerRevisions of VirtualMachineFlavorSpec and VirtualMachinePreferenceSpec", func() {
+
+			var (
+				flavorRevision     *appsv1.ControllerRevision
+				preferenceRevision *appsv1.ControllerRevision
+			)
+
+			vmi := tests.NewRandomVMI()
+
+			By("Creating a VirtualMachineFlavor")
+			flavor := newVirtualMachineFlavor(vmi)
+			originalFlavorCPUGuest := flavor.Spec.CPU.Guest
+			flavor, err := virtClient.VirtualMachineFlavor(util.NamespaceTestDefault).
+				Create(context.Background(), flavor, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Creating a VirtualMachinePreference")
+			preference := newVirtualMachinePreference()
+			preference.Spec = flavorv1alpha1.VirtualMachinePreferenceSpec{
+				CPU: &flavorv1alpha1.CPUPreferences{
+					PreferredCPUTopology: flavorv1alpha1.PreferSockets,
+				},
+			}
+			preference, err = virtClient.VirtualMachinePreference(util.NamespaceTestDefault).
+				Create(context.Background(), preference, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Creating a VirtualMachine")
+			removeResourcesAndPreferencesFromVMI(vmi)
+			vm := tests.NewRandomVirtualMachine(vmi, false)
+
+			vm.Spec.Flavor = &v1.FlavorMatcher{
+				Name: flavor.Name,
+				Kind: flavorapi.SingularResourceName,
+			}
+			vm.Spec.Preference = &v1.PreferenceMatcher{
+				Name: preference.Name,
+				Kind: flavorapi.SingularPreferenceResourceName,
+			}
+
+			vm, err = virtClient.VirtualMachine(util.NamespaceTestDefault).Create(vm)
+			Expect(err).ToNot(HaveOccurred())
+
+			vm = tests.StartVirtualMachine(vm)
+
+			expectedFlavorRevisionName := flavorpkg.GetRevisionName(vm.Name, flavor.Name, flavor.UID, flavor.Generation)
+			By("Waiting for a VirtualMachineFlavorSpec ControllerRevision to be referenced from the VirtualMachine")
+			Eventually(func() bool {
+				vm, err = virtClient.VirtualMachine(vm.Namespace).Get(vm.Name, &metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return (vm.Spec.Flavor.RevisionName == expectedFlavorRevisionName)
```

Comment:
Can you instead return `vm.Spec.Flavor.RevisionName` from the function and then check it using `.Should(Equal(expectedFlavorRevisionName))` ?